src/lib/utils/concepts.h

```diff
@@ -29,6 +29,18 @@ struct is_strong_type<Strong<Ts...>> : std::true_type {};
 template <typename... Ts>
 constexpr bool is_strong_type_v = is_strong_type<std::remove_const_t<Ts>...>::value;
 
+/**
+ * Checks whether a strong type has the @p Capability included in its @p Tags type pack.
+ */
+template <typename Capability, typename T, typename... Tags>
+constexpr auto strong_type_has_capability(Strong<T, Tags...>) {
+   if constexpr((std::is_same_v<Capability, Tags> || ...)) {
+      return std::true_type();
+   } else {
+      return std::false_type();
+   }
+}
```

Comment:
And another architecture suggestion: Why not let the `Strong<>` template tell you what it can do via a static template method in `class Strong<>`:

```C++
      template <typename CapabilityT>
      constexpr static bool has_capability() {
         return (std::is_same_v<CapabilityT, Capabilities> || ...);
      }
```

See here for a quick sketch: https://godbolt.org/z/jxebM7fvK

Reply:
Nice, it works! Thanks!